### DISCUSSION TITLE
How to call class methods from a custom Field compoent?

### DISCUSSION THREAD

**Discussion by zabaala** [filamentphp/filament#8159] [Help]
Hi, everyone.

I'm working on creation of a custom field which need to call a function from own component class. Something like a tick, by example.

```blade
<div>
    <a wire:click="tick()">Tick</a>
</div>
```  

```php

class CustomField extend Field
{
  public function tick()
  {
    debug('ticked');
  }
}
```

But this approach cause error when I try to call `tick()` from the view, with message:

```
Livewire\Exceptions\MethodNotFoundException:
Unable to call component method. Public method [tick] not found on component
```

What I need to do to solve it?

Thanks.

**Comment by leandrocfe** [comment]
Filament v3 supports actions
https://github.com/filamentphp/filament/discussions/8153#discussioncomment-6853760

**Reply by zabaala** [reply]
Actions not sounds good for me for this case, @leandrocfe.
I'm looking for something which allow me to call a component class method.

**Comment by danharrin** [comment]
Form components are not Livewire components.

**Reply by zabaala** [reply]
Hmmm...
So... how internally this works? Because I saw calls like $wire.something() in Repeater and in Table, by example. In this case, the right way is create a livewire component to call inside a custom field or something like that?

**Reply by danharrin** [reply]
Repeater does not use $wire or wire:click

You should use form component actions - https://filamentphp.com/docs/3.x/forms/actions#adding-an-action-to-a-custom-form-component

Instead of calling `registerActions()` on a ViewField, you could do `$this->registerActions()` in the `setUp()` method of your component class

**Comment by BenQoder** [comment]
So is there a workaround to call something like $wire.generateUuid()?

generateUuid is a public function in the Field class

**Reply by danharrin** [reply]
https://filamentphp.com/docs/3.x/forms/actions#adding-an-action-to-a-custom-form-component

**Reply by ahmic** [reply]
As a workaround for calling a function from a component class, you can register an action as mentioned above. Then, you can invoke it from Alpine like this

`$wire.mountFormComponentAction('{{ $getStatePath() }}', 'generateUuid');`
